cloud/services/groups/service.go

```diff
@@ -30,6 +30,6 @@ type Service struct {
 func NewService(scope *scope.ClusterScope) *Service {
 	return &Service{
 		Scope:  scope,
-		Client: NewClient(scope.SubscriptionID, scope.Authorizer),
+		Client: NewClient(scope.SubscriptionID(), scope.Authorizer()),
```

Comment:
```
NewClient(scope.SubscriptionID(), scope.Authorizer())
``` 
is commonly used. Would it make sense to change refactor those methods into an interface and pass an instance of that interface, like 
```
NewClient(authorizer Authorizer)
```

Reply:
I have to rebase this on top of #649 which will remove SubID and add settings, will see what I can with that after the rebase

Reply:
Done, let me know what you think

Reply:
Looks good.

Do you feel like the interface fits or should it be a struct?

Reply:
I like how you created a wrapper Authorizer to make the func signature simpler

Reply:
I just pushed a new update @nader-ziada @devigned. Removed Settings() and replaced it with SubscriptionID() and BaseURI()

Reply:
The interface fits I think because I pass in scope to NewClient (scope implements azure.Authorizer)

Reply:
yeah makes sense now that you have `Authroizer` as a type wrapping all the needed fields